RIOT-OS/RIOT
# pkg/tinyusb: fix USB speed selection for STM32 boards with HS PHY

### Contribution description

This PR again fixes the USB speed selection for STM32 boards with HS PHY, which was still incorrect with PR #18906. It fixes also the problem after `CLOCK_HSE` has been renamed `CONFIG_CLOCK_HSE`.

If the HS port of the STM32 is used together with a HS-PHY enabled by the module `periph_usbdev_hs_utmi` or `periph_usbdev_hs_ulpi`, `OPT_MODE_HIGH_SPEED` must be set as `CFG_TUSB_RHPORT1_MODE`. Otherwise the built-in on-chip FS-PHY is used and `OPT_MODE_DEFAULT_SPEED` should be set as `CFG_TUSB_RHPORT1_MODE`. 

Using `CFG_TUD_MAX_SPEED` for this selection, as provided in PR #18906, works for `stm32f429i-disc1` where the HS port is used with the on-chip FS PHY, but not for boards such as `stm32f746g-disco` or `stm32f723e-disco` where the HS port is used with an HS PHY.

### Testing procedure

The HS port should work for all three mentioned boards with this PR:
```
BOARD=stm32f429i-disc1 make -j8 -C tests/pkg_tinyusb_cdc_msc flash
```
```
USEMODULE=periph_usbdev_hs_utmi BOARD=stm32f723e-disco make -j8 -C tests/pkg_tinyusb_cdc_msc flash
```
```
USEMODULE=periph_usbdev_hs_ulpi BOARD=stm32f746g-disco make -j8 -C tests/pkg_tinyusb_cdc_msc flash
```
Without this PR, the test works for `stm32f429i-disc1` but not for `stm32f723e-disco` and `stm32f746g-disco`

### Issues/PRs references

Fixes PR #18906
